### ISSUE TITLE
Detroying aws_route53_zone_association fails due to nonexisting hosted zone

### ISSUE THREAD

**Issue by traittinen** [hashicorp/terraform-provider-aws#17016]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

Terraform v0.14.0
AWS Provider v3.22.0

### Affected Resource(s)

* aws_route53_zone_association

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
# ACCOUNT A

resource "aws_route53_zone" "aws_dns" {
  name = "itest.aws.dnstest.intra"

  vpc {
    vpc_id = module.vpc.vpc_id
  }

  lifecycle {
    ignore_changes = [vpc]
  }

  tags = {
    Name = "itest.aws.dnstest.intra"
  }
}

resource "aws_route53_vpc_association_authorization" "aws_dns" {
  vpc_id  = data.terraform_remote_state.shared-services.outputs.vpc-shared_services
  zone_id = aws_route53_zone.aws_dns.id
}

# ACCOUNT B

resource "aws_route53_zone_association" "cloud2-tomi-sandbox2" {
  count = length(data.terraform_remote_state.cloud2-tomi-sandbox2.outputs.shared_zones)
  zone_id = data.terraform_remote_state.cloud2-tomi-sandbox2.outputs.shared_zones[count.index]
  vpc_id  = module.vpc.vpc_id
}
```

### Debug Output

Destroying the resource fails on disassociating the zone because zone does not exists anymore:
https://gist.github.com/traittinen/3ea453c3897bee3dbf02bd462e4e475a

### Expected Behavior

If zone/association does not exist. Resource should be deleted from state on destroy and execution should not be aborted.

### Actual Behavior

Terraform destroy fails because zone doesn't exist.

### Steps to Reproduce

1. Create aws_route53_zone on account A VPC X
2. Share zone (aws_route53_vpc_association_authorization) with VPC Y (owned by account B) on account A.
3. Associate the hosted zone (aws_route53_zone_association) with VPC Y on account B.
4. Delete zone on account A
5. Delete zone association on account B

### Important Factoids

This would probably work when executing steps 4 and 5 in reverse order, but since accounts may be handled by different parties the order of deleting resources is not always guaranteed.

### References

* #0000

**Comment by bflad** [comment]
Hi @traittinen 👋  Thank you for reporting this and sorry you ran into trouble here. A fix for this has been submitted here: https://github.com/hashicorp/terraform-provider-aws/pull/17023

**Comment by bflad** [comment]
The fix for this has been merged and will be released with version 3.24.0 of the Terraform AWS Provider, later this week. 👍
